backend/app/utils.py

```diff
@@ -0,0 +1,32 @@
+from yarl import URL
+
+
+def make_dsi(dataspace_base_domain: str, definition_path: str, source: str) -> str:
+    """
+    Construct a Data Source Identifier from the information on the request and token
+
+    DSIs are URIs in the format: dpp://<source>@<dataspace_base_domain>/<data_definition>
+    """
+
+    # Data sources are identified on dataspaces as <group> or <group>:<variant>, which on the DSI URI correspond to a
+    # "user" and an optional "password"
+    user, _, password = source.partition(":")  # Extract group:source to URL properties
+
+    if not password:
+        # Passing the value None will not append it to the URL, whereas an empty string would be
+        password = None
+
+    # Use a URL builder to build the URI correctly
+    #
+    # Our expectation is that all the values are suitable for doing this with just string concatenation, however for
+    # future proofing we want to do this a bit more carefully here.
+
+    dsi_url = URL.build(
+        scheme="dpp",
+        host=dataspace_base_domain,
+        path=definition_path,
```

Comment:
How about changing this helper function to take in the definition instead of the definition path?
You could then on this line add the leading slash (`/`) to convert it to a path and then you could simplify the code in both places where you call this helper.

Reply:
Then it would have to be also changed in the `example-productizer`, I intentionally kept this as an exact copy

Reply:
I'll leave it up to you to consider if you want to keep it as it is, change it to check if there's already a slash or not and add it if needed or if you want to make the functions across the repos be out of sync.

Reply:
Made it to automatically add the prefix if missing, and another PR to the example-productizer